broker/tasks/pipelines.py

```diff
@@ -282,6 +276,17 @@ def queue_all_alb_update_tasks_for_operation(operation_id, correlation_id):
         .then(alb.select_alb, operation_id, correlation_id=correlation_id)
         .then(alb.add_certificate_to_alb, operation_id, correlation_id=correlation_id)
         .then(route53.create_ALIAS_records, operation_id, correlation_id=correlation_id)
+        .then(route53.wait_for_changes, operation_id, correlation_id=correlation_id)
+        .then(
+            alb.remove_certificate_from_previous_alb,
+            operation_id,
+            correlation_id=correlation_id,
+        )
+        .then(
+            iam.delete_previous_server_certificate,
+            operation_id,
+            correlation_id=correlation_id,
+        )
```

Comment:
Based on the flow here, it's wait until the ALB is ready to be modified, then remove the certificate from the ALB, then delete the certificate itself, correct?

Reply:
right - we need to wait for the route53 changes to propagate so we're sure we can remove the cert from the old ALB. Then we want to remove the cert from the ALB before removing it from IAM. (we actually could remove it from IAM first, then ALB second, but it's not recommended and logs some warnings, in addition to not making any sense to do)

Reply:
Fair enough!  And you're right, it doesn't make sense to do it in reverse order.

Thank you for explaining, I appreciate it!